### PR TITLE
[ticket/15884] Add memberlist_body_* events

### PR DESCRIPTION
**Pull request by 3D-I**
PHPBB3-15884

Checklist:

- [x] Correct branch: master for new features; 3.2.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.2.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-15884

### CODE REVIEW

**Comment by toxyy** [comment]
Perhaps it doesn't like mixing in twig syntax?

**Comment by 3D-I** [comment]
https://github.com/phpbb/phpbb/pull/5386

**Comment by mrgoldy** [comment]
Also, please note that all these events are within the table.
If an extension developer wants to show some additional group information, it would require an event/few events in [this block](https://github.com/phpbb/phpbb/blob/7a92f6a3b63bf4de7cd9f0d4854ef981c7444465/phpBB/styles/prosilver/template/memberlist_body.html#L16-L26)

Might look something like this, following the [user profile page layout](https://github.com/phpbb/phpbb/blob/master/phpBB/styles/prosilver/template/memberlist_view.html):
```html
<!-- IF S_SHOW_GROUP -->
	<!-- EVENT memberlist_body_group_name_before -->
	<h2 class="group-title"<!-- IF GROUP_COLOR --> style="color:#{GROUP_COLOR};"<!-- ENDIF -->>{GROUP_NAME}</h2>
	<!-- EVENT memberlist_body_group_name_after -->
	<!-- IF U_MANAGE -->
		<p class="right responsive-center manage rightside"><a href="{U_MANAGE}">{L_MANAGE_GROUP}</a></p>
	<!-- ENDIF -->
	<p>{GROUP_DESC} {GROUP_TYPE}</p>

	<!-- EVENT memberlist_body_group_desc_after -->

	<p>
		<!-- IF AVATAR_IMG -->{AVATAR_IMG}<!-- ENDIF -->
		<!-- EVENT memberlist_body_group_rank_before -->
		<!-- IF RANK_IMG -->{RANK_IMG}<!-- ENDIF -->
		<!-- IF GROUP_RANK -->{GROUP_RANK}<!-- ENDIF -->
		<!-- EVENT memberlist_body_group_rank_after -->
	</p>
<!-- ELSE -->
```

Moreover, the 'group rank' (`{GROUP_RANK}` && `{RANK_IMG}`) will never show up on that page.
As the group data is called [here](https://github.com/phpbb/phpbb/blob/master/phpBB/memberlist.php#L1215), with the `group_` prefix and send off to `phpbb_get_user_rank()` which uses the `user_` prefix, see [here](https://github.com/phpbb/phpbb/blob/master/phpBB/includes/functions_display.php#L1512). But I suppose that's a different issue.

**Comment by 3D-I** [comment]
It's ok by, my side, to add those events as per your above note. I will.
What I did till now is to add events related to other kind of infos in the table.

About the issue as per below:

> Moreover, the 'group rank' (`{GROUP_RANK}` && `{RANK_IMG}`) will never show up on that page.
> As the group data is called [here](https://github.com/phpbb/phpbb/blob/master/phpBB/memberlist.php#L1215), with the `group_` prefix and send off to `phpbb_get_user_rank()` which uses the `user_` prefix, see [here](https://github.com/phpbb/phpbb/blob/master/phpBB/includes/functions_display.php#L1512). But I suppose that's a different issue.

Ofcourse it is another stuff, I see you are working about thatin fact:

https://tracker.phpbb.com/browse/PHPBB3-15885
https://tracker.phpbb.com/browse/PHPBB3-15886

Thx for feedback. :+1:

**Comment by 3D-I** [comment]
Ok, after that we are done here. :smiley_cat:

File: phpBB/docs/events.md [hunk 1]
```diff
@@ -1114,6 +1114,76 @@ mcp_warn_user_add_warning_field_before
...
+* Purpose: Add data after the page header.
+
+memberlist_body_page_title_before
+===
+* Locations:
+    + styles/prosilver/template/memberlist_body.html
+* Since: 3.2.6-RC1
+* Purpose: Add data before the page title.
+
+memberlist_body_group_name_before
```
Comment (marc1706) [inline phpBB/docs/events.md]: If sorting alphabetically, memberlist_body_group_name_before should definitely go before memberlist_body_page_title_before.

Comment (marc1706) [reply phpBB/docs/events.md]: The rest is also not sorted alphabetically. Please try to always sort newly added events alphabetically.

File: phpBB/docs/events.md [hunk 2]
```diff
@@ -1146,6 +1216,20 @@ memberlist_body_username_prepend
 * Purpose: Add information before every username in the memberlist. Works in
 all display modes (leader, group and normal memberlist).
 
+memberlist_body_memberrow_after
```
Comment (marc1706) [inline phpBB/docs/events.md]: This also does not fit in the correct order

**Comment by 3D-I** [comment]
!unset Styles 🎨

**Comment by 3D-I** [comment]
!set Event

**Pull request merged**